Add App Hangs documentation for RN SDK (#6951)
getsentry/sentry-docs: src/platforms/react-native/configuration/app-hangs.mdx

---
title: App Hangs
sidebar_order: 11
description: "Learn about how to add app hang detection reporting."
---

This integration tracks app hangs. This feature is available on iOS, tvOS, and macOS.

Trying to use an unresponsive app is extremely frustrating for users. There are many reasons why an app may become unresponsive, such as long-running code, infinite loop bugs, and so on. With app hang tracking you can detect and fix them.

Every time App Hang is detected an error event is created. App Hang Tracking is enabled by default. To disable it, set the `enableAppHangTracking` option to `false`:

```javascript
import * as Sentry from '@sentry/react-native';

Sentry.init({
  dsn: 'DSN',
  enableAppHangTracking: false,
});
```

> The app hang detection integration has a default timeout of two (2) seconds. It represents the minimum amount of time an app should be unresponsive to be classified as an App Hanging. You can change the timeout by changing the `appHangTimeoutInterval` option:

Reviewer: ```suggestion
The app hang detection integration has a default timeout of two (2) seconds. This represents the minimum amount of time an app can be unresponsive before it is classified as hanging. You can change the timeout by setting the `appHangTimeoutInterval` option:
```

```javascript
import * as Sentry from '@sentry/react-native';

Sentry.init({
  dsn: 'DSN',
  appHangTimeoutInterval: 1,
});
```

> To read more about how app hangs are detected, check out the `sentry-cocoa` [documentation](/platforms/apple/app-hangs/).

Reviewer: ```suggestion
To read more about how app hangs are detected, check out the `sentry-cocoa` [documentation](/platforms/apple/configuration/app-hangs/).
```

Reviewer: Looks like this slug was wrong
